# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/mubashir-dev751/starter/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [ ]:
I am choosing Lane 2: Refresh / Content Opportunity Scoring.
A model is never the goal; a better decision is. The business problem here is a lack of editorial capacity, not a lack of data. The system currently uses static baseline rules which flag too many pages, creating an unmanageable queue. By scoring and ranking these opportunities, we transition from a noisy "rule-based alert" system to a prioritized decision-support tool.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [ ]:
The Decision: Which declining pages should human editors review first to maximize potential traffic recovery?

The Action: A human reviewer inspects the top-ranked pages to trigger an update, rewrite, or metadata optimization.

Cost of a Wrong Call: Wasted editorial hours spent rewriting a page that dropped due to uncontrollable factors (like seasonality or market consolidation), while a high-value, genuinely decaying page is ignored.

Why ML helps: Static rules flag every drop equally. ML earns its place because the pattern of true decay is tangled across multiple signals (age, impression volume, position tiers, and engagement rates) that are too messy to write by hand.

The Frame: For human editors, deciding which declining pages to review first, we will build a ranked queue from observable search and content signals, scoring decay risk measured by Precision@50. A wrong call costs wasted editorial hours. A plain rule isn't enough because it flags too many false positives. We will claim only decision-support results.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [11]:
import pandas as pd

df = pd.read_csv('content_refresh_anonymized.csv')

print(f"Total pages: {len(df)}")
print(f"Declining pages: {len(df[df['trend_direction'] == 'down'])}")
print(f"High volume decline (>500 impressions): {len(df[(df['trend_direction'] == 'down') & (df['impressions_90d'] > 500)])}")

Total pages: 30000
Declining pages: 16262
High volume decline (>500 impressions): 9956


The data justifies this project. Out of 30,000 pages in the starter dataset, 16,262 are marked as declining. Crucially, 9,956 of those have significant traffic volume (>500 impressions). A human cannot manually review almost 10,000 pages; ML is required to rank this high-volume subset effectively.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [ ]:
What I can claim: I can rank pages that share observational signals with historical traffic decay to prioritize human review.

What I cannot claim: I cannot claim causality. I cannot guarantee that refreshing a high-ranking page will recover its traffic, and I cannot isolate Google's algorithm factors.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.